src/index.ts

```diff
@@ -65,14 +74,10 @@ export interface DagJWSResult {
   linkedBlock: Uint8Array
 }
 
-export interface ResolverOptions {
-  registry?: ResolverRegistry
-  cache?: DIDCache | boolean
-}
-
 export interface DIDOptions {
   provider?: DIDProvider
-  resolver?: Resolver | ResolverOptions
+  resolver?: Resolver | ResolverRegistry
+  cache?: DIDCache
```

Comment:
I'd rather keep the types simple than overspecify them. Will be harder for ppl to understand what's going on I think

Reply:
I think it limits the advantage of using TS then. I find it useful to have compilation errors/warnings when setting invalid options, rather than something too permissive that might lead to wrong expectations.

Reply:
I think there is a balance to be struck here. I think the main advantage of types is that it becomes easier for devs to read and understand the code.